xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R140300001XWIKI19571DataMigration.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.index.migration;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.engine.jdbc.connections.spi.JdbcConnectionAccess;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.mapping.PersistentClass;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.doc.tasks.XWikiDocumentIndexingTask;
+import org.xwiki.index.internal.TasksStore;
+
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.internal.store.hibernate.HibernateStore;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+import com.xpn.xwiki.store.migration.hibernate.AbstractHibernateDataMigration;
+
+/**
+ * Migrate XWikiDocumentIndexingTask by copying the table to a new temporary table. Then, after hibernate migration,
+ * when XWikiDocumentIndexingTask is created again, the rows from the temporary table are copied to the new table.
+ * Finally, the temporary table is dropped.
+ *
+ * @version $Id$
+ * @since 14.3RC1
+ * @since 14.2.1
+ */
+@Component
+@Singleton
+@Named(R140300001XWIKI19571DataMigration.HINT)
+public class R140300001XWIKI19571DataMigration extends AbstractHibernateDataMigration
```

Comment:
While this is working correctly with hsqldb, I couldn't test it with Oracle yet. I'll update the corresponding test (and the coverage) once I'll manage to test this.